hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/wal/AbstractFSWAL.java

```diff
@@ -986,14 +996,43 @@ public void shutdown() throws IOException {
         i.logCloseRequested();
       }
     }
-    rollWriterLock.lock();
+
+    ExecutorService shutdownExecutor = Executors.newFixedThreadPool(1,
```

Comment:
This will be the second single thread ExecutorService in AbstractFSWAL, can we do it optimized way?

Reply:
Thanks. The WAL shutdown task should have priority to be executed, it may be delayed if put to the same single thread pool as archive WAL. What do you think?

Reply:
Hi, @pankaj72981 , I have merged the archiver and shutdown executor, please take a look, thanks.